src/components/TranslationDialog/TranslationModal/TranslationForm.js

```diff
@@ -0,0 +1,138 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    CenteredContent,
+    DataTable,
+    DataTableBody,
+    DataTableCell,
+    DataTableColumnHeader,
+    DataTableHead,
+    DataTableRow,
+    Input,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { LocalesSelect } from './LocalesSelect.js'
+
+export const TranslationForm = ({
+    fieldsToTranslate,
+    objectToTranslate,
+    translations,
+    onUpdateTranslations,
+}) => {
+    const [newTranslations, setNewTranslations] = useState(translations)
+    const [translationLocale, setTranslationLocale] = useState()
+
+    const camelCaseToUnderscores = field =>
+        field
+            .replace(/[a-z][A-Z]/g, match =>
+                [match.chartAt(0), match.chartAt(1)].join('_')
+            )
+            .toLowerCase()
+
+    const getTranslationIndexForField = field => {
+        console.log(
+            'translations',
+            newTranslations,
+            camelCaseToUnderscores(field)
+        )
+        return newTranslations.findIndex(
+            element =>
+                element.locale === translationLocale &&
+                element.property.toLowerCase() === camelCaseToUnderscores(field)
+        )
+    }
+
+    const getTranslationForField = field => {
+        const translationIndex = getTranslationIndexForField(field)
+        console.log(field, translationIndex)
+        return translationIndex !== -1
+            ? newTranslations[translationIndex]?.value || ''
+            : ''
+    }
+
+    const setTranslationForField = (field, translation) => {
+        console.log('setTranslationField called', field, translation)
+
+        const newTranslation = {
+            locale: translationLocale,
+            property: camelCaseToUnderscores(field).toUpperCase(),
+            value: translation,
+        }
+
+        const translationIndex = getTranslationIndexForField(field)
+
+        if (translationIndex !== -1) {
+            newTranslations.splice(translationIndex, 1, newTranslation)
+        } else {
+            newTranslations.push(newTranslation)
```

Comment:
65 and 67: do not modify the state variable

Reply:
Yep, figured it didn't work, fixed.